govcd/system.go

```diff
@@ -951,6 +940,20 @@ func (vcdClient *VCDClient) QueryNsxtManagerByName(name string) ([]*types.QueryR
 	return results.Results.NsxtManagerRecord, nil
 }
 
+// QueryNsxtManagerByHref searches for NSX-T managers available in VCD
+func (vcdClient *VCDClient) QueryNsxtManagerByHref(href string) ([]*types.QueryResultNsxtManagerRecordType, error) {
+	results, err := vcdClient.QueryWithNotEncodedParams(nil, map[string]string{
+		"type":          "nsxTManager",
+		"filter":        fmt.Sprintf("href==%s", url.QueryEscape(extractUuid(href))),
```

Comment:
So should the  HREF be equal to UUID in this case?

Reply:
In most cases (if not all), the query system extracts the UUID internally, and uses it for comparison. 
In this case, when the NSX-T manager HREF is used in the provider VDC, its HREF is too long and can trigger an error. For this reason I preferred extracting the UUID before using as a parameter